### PR TITLE
Implement audio using SoLoud

### PR DESCRIPTION
**Pull request by Unit158**
The current test case sucks, and I've only tested building this on windows.

### CODE REVIEW

File: src/asset_manager.cc [hunk 1]
```diff
@@ -138,6 +141,29 @@ void asset_manager::load_asset_manifest(char const *filename) {
...
+            config_setting_lookup_string(asset_setting, "file", &asset_file);
+            int streaming;
+            config_setting_lookup_bool(asset_setting, "streaming", &streaming);
+
+            printf("\n%s %s %i\n", asset_name, asset_file, streaming);
+
+            if (streaming) {
+                SoLoud::WavStream * sound = new SoLoud::WavStream();
+                sounds[asset_name] = sound;
+                printf("\n\n\n%i\n\n\n", sound->load(asset_file));
```
Comment (Unit158) [inline src/asset_manager.cc]: Forgot to remove this and the statement below.

**Comment by RobotCaleb** [comment]
I don't hear anything on Linux. Should I?

**Comment by RobotCaleb** [comment]
It doesn't build for me with ALSA, some undefined reference
It does build with OSS, but still no sound for me.

```
/usr/bin/ld: libNIGHTMARE.a(soloud_alsa.cpp.o): undefined reference to symbol 'snd_pcm_hw_params_any@@ALSA_0.9'
//usr/lib/x86_64-linux-gnu/libasound.so.2: error adding symbols: DSO missing from command line
```

File: main.cc [hunk 2]
```diff
@@ -293,6 +296,10 @@ init()
 
     // Absorb all the init time so we dont try to catch up
     frame_info.tick();
+
+    audio = new SoLoud::Soloud();
```
Comment (chrisforbes) [inline main.cc]: init before the tick() call just above -- otherwise sim has to catch up all the time the init takes.

**Comment by Unit158** [comment]
Updated.

**Comment by chrisforbes** [comment]
@Unit158 what needs to happen to get this landable?

**Comment by Unit158** [comment]
I forgot to push my changes. oops

**Comment by Unit158** [comment]
Updated.

**Comment by Unit158** [comment]
reminder: the handling of the threading of the audio sources should be updated upstream.

**Comment by RobotCaleb** [comment]
@Unit158 Thanks. Merged.

**Pull request merged**